### PR TITLE
Deactivate webhook before deleting

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because it prevents issue where using `WebhookDelete` mutation while tasks that create `EventDeliveryAttempt` are queued resulted in `IntegrityError`.

The most simple resolution was to deactivate Webhook before deleting it in mutation. `send_webhook_request_async` task is checking if webhook `is_active` so this should prevent this situation.

Tasks:
I want to add tests after a discussion regarding edge-case.

Things to discuss: 
⚠️ There is a small window for this solution to fail - when Webhook deactivation will happen between the `is_active` check and attempt creating/updating in `send_webhook_request_async` task. I'm not quite sure how to handle this case. Should I add more webhook activity checks inside `send_webhook_request_async` task or maybe catch this edge-case exception to fail gracefully?  Any ideas? 

Second thing: Do we want to do the sam analogically for deleting App? (deactivate webhook in app deletion)

#11227 
# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by SzymJ — APPROVED** [review]

File: saleor/graphql/webhook/mutations.py [hunk 1]
```diff
@@ -255,14 +256,22 @@ def perform_mutation(cls, _root, info, **data):
...
             except models.Webhook.DoesNotExist:
                 raise ValidationError(
                     f"Couldn't resolve to a node: {node_id}",
                     code=WebhookErrorCode.GRAPHQL_ERROR,
                 )
-
-        return super().perform_mutation(_root, info, **data)
+        try:
+            response = super().perform_mutation(_root, info, **data)
+        except IntegrityError:
```
Comment (przlada) [inline saleor/graphql/webhook/mutations.py:269]: You are only deactivating the webhook if the delete mutation comes from an app, not the user

**Review by przlada — CHANGES_REQUESTED** [review]

File: saleor/graphql/webhook/tests/test_webhook.py [hunk 2]
```diff
@@ -305,6 +305,24 @@ def test_webhook_delete_by_inactive_app(app_api_client, webhook):
     assert_no_permission(response)
 
 
+@patch("saleor.webhook.models.Webhook.delete")
+def test_webhook_delete_deactivates_before_deletion(
```
Comment (korycins) [inline saleor/graphql/webhook/tests/test_webhook.py:309]: Wouldn't be good to have a separate test, that will check that ValidationError has been raised?

File: saleor/graphql/webhook/mutations.py [hunk 3]
```diff
@@ -248,21 +249,32 @@ def perform_mutation(cls, _root, info, **data):
...
+                webhook = app.webhooks.get(id=object_id)
             except models.Webhook.DoesNotExist:
                 raise ValidationError(
                     f"Couldn't resolve to a node: {node_id}",
                     code=WebhookErrorCode.GRAPHQL_ERROR,
                 )
-
-        return super().perform_mutation(_root, info, **data)
+        try:
+            webhook = webhook or models.Webhook.objects.get(object_id)
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/webhook/mutations.py:267]: WDYT?

[the remark this answers is not in the document]
Comment (jakubkuc) [reply saleor/graphql/webhook/mutations.py:267]: Brilliant, I'll test it

File: saleor/graphql/webhook/mutations.py [hunk 4]
```diff
@@ -248,21 +249,32 @@ def perform_mutation(cls, _root, info, **data):
...
+            webhook.is_active = False
+            webhook.save(update_fields=["is_active"])
+            response = super().perform_mutation(_root, info, **data)
+        except IntegrityError:
+            raise ValidationError(
+                "Webhook couldn't be deleted at this time due to running task."
+                "Webhook deactivated. Try deleting Webhook later",
+                code=WebhookErrorCode.GRAPHQL_ERROR,
```
[the remark this answers is not in the document]
Comment (jakubkuc) [reply saleor/graphql/webhook/mutations.py:275]: Good point. I'll add a new code. Maybe `DELETE_FAILED`?

File: saleor/graphql/webhook/mutations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakubkuc) [reply saleor/graphql/webhook/mutations.py]: Works well. I've updated the PR 🚀

**Review by fowczarek — APPROVED** [review]

**Review by przlada — APPROVED** [review]

File: saleor/graphql/webhook/mutations.py [hunk 5]
```diff
@@ -244,25 +245,31 @@ class Meta:
...
-        return super().perform_mutation(_root, info, **data)
+        if app and not app.is_active:
+            raise ValidationError(
+                "App needs to be active to delete webhook",
+                code=WebhookErrorCode.INVALID,
+            )
+        webhook = cls.get_node_or_error(info, data.get("id"), only_type=Webhook)
+        if app and webhook.app.id != app.id:
+            raise ValidationError(
+                f"Couldn't resolve to a node: {webhook.app.id}",
```
Comment (maarcingebala) [inline saleor/graphql/webhook/mutations.py:257]: We should return GraphQL node ID, but `webhook.app.id` is a database ID. Also before it was returning the ID of the webhook, not the app.

File: saleor/graphql/webhook/mutations.py [hunk 6]
```diff
@@ -244,25 +245,31 @@ class Meta:
...
-                )
-
-        return super().perform_mutation(_root, info, **data)
+        if app and not app.is_active:
+            raise ValidationError(
+                "App needs to be active to delete webhook",
+                code=WebhookErrorCode.INVALID,
+            )
+        webhook = cls.get_node_or_error(info, data.get("id"), only_type=Webhook)
+        if app and webhook.app.id != app.id:
```
Comment (maarcingebala) [inline saleor/graphql/webhook/mutations.py:255]: Doesn't `webhook.app.id` make a database query? Could be `webhook.app_id` instead I guess.

**Review by maarcingebala — COMMENTED** [review]
We need to update the description of `webhookDelete` mutation to explain that the deletion might fail if there is a webhook delivery in progress, but webhooks get deactivated in that case.
